hazelcast/src/main/java/com/hazelcast/client/impl/protocol/ClientProtocolErrorCodes.java

```diff
@@ -0,0 +1,237 @@
+            @Override
+            public Throwable createException(String message) {
+                return new AuthenticationException(message);
+            }
+        });
+        register(HAZELCAST_INSTANCE_NOT_ACTIVE, HazelcastInstanceNotActiveException.class, new ExceptionFactory() {
+            @Override
+            public Throwable createException(String message) {
+                return new HazelcastInstanceNotActiveException(message);
+            }
+        });
+        register(ILLEGAL_STATE, IllegalStateException.class, new ExceptionFactory() {
+            @Override
+            public Throwable createException(String message) {
+                return new IllegalStateException(message);
+            }
+        });
+        register(ILLEGAL_ARGUMENT, IllegalArgumentException.class, new ExceptionFactory() {
+            @Override
+            public Throwable createException(String message) {
+                return new IllegalArgumentException(message);
+            }
+        });
+        register(EXECUTION, ExecutionException.class, new ExceptionFactory() {
+            @Override
+            public Throwable createException(String message) {
+                return new ExecutionException(message, null);
+            }
+        });
+        register(TIMEOUT, TimeoutException.class, new ExceptionFactory() {
+            @Override
+            public Throwable createException(String message) {
+                return new TimeoutException(message);
+            }
+        });
+        register(INTERRUPTED, InterruptedException.class, new ExceptionFactory() {
+            @Override
+            public Throwable createException(String message) {
+                return new InterruptedException(message);
+            }
+        });
+        register(ILLEGAL_MONITOR_STATE, IllegalMonitorStateException.class, new ExceptionFactory() {
+            @Override
+            public Throwable createException(String message) {
+                return new IllegalMonitorStateException(message);
+            }
+        });
+        register(DISTRIBUTED_OBJECT_DESTROYED, DistributedObjectDestroyedException.class, new ExceptionFactory() {
+            @Override
+            public Throwable createException(String message) {
+                return new DistributedObjectDestroyedException(message);
+            }
+        });
+        register(ILLEGAL_STATE, IllegalStateException.class, new ExceptionFactory() {
+            @Override
+            public Throwable createException(String message) {
+                return new IllegalStateException(message);
+            }
+        });
+        register(HAZELCAST, HazelcastException.class, new ExceptionFactory() {
+            @Override
+            public Throwable createException(String message) {
+                return new HazelcastException(message);
+            }
+        });
+        //TODO following list will be registered !?!?!
```

Comment:
We will register all 64 exceptions that we use now. When we start to use a new exception that is not in the list, it will be send as HazelcastException. Are we OK with that ?